api/v1alpha4/common_types.go

```diff
@@ -53,6 +53,12 @@ const (
 	// on the reconciled object.
 	PausedAnnotation = "cluster.x-k8s.io/paused"
 
+	// WatchLabel is a label othat can be applied to any Cluster API object.
+	//
+	// Controllers which allow for selective reconciliation may check this label and proceed
+	// with reconciliation of the object only if this label and a configured value is present.
+	WatchLabel = "cluster.x-k8s.io/watch-filter"
```

Comment:
Should we turn this into a prefix rather than having watches based on key + values?
```suggestion
	WatchLabelPrefix = "cluster.x-k8s.io/watch"
```
It could be used as a label like: "cluster.x-k8s.io/watch/test"

Reply:
I would be fine doing it either way. Key + value seems a little bit more intuitive to me but I am open to change it.

Reply:
what would the UX look like if using a prefix + suffix? You'd still have to pass in the suffix in the controller, and then you need to make sure you're setting the label the right way, ie. `cluster.x-k8s.io/watch/test` and not `cluster.x-k8s.io/watch-test` or `cluster.x-k8s.io/watch_test`.

IMO, key + value is more intuitive as @MarcelMue said.